### PR TITLE
docs: add page on abigen

### PR DESCRIPTION
**Pull request by jmcook1186**
adds a short page outlining basic usage of abigen.

### CODE REVIEW

**Review by MariusVanDerWijden — COMMENTED** [review]
Very cool! Thanks for writing this, added a few nitpicks

File: docs/_dapp/abigen.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+}
+...
+
+```
+
+`Storage.go` contains all the bindings required to interact with 
+`Storage.sol` from a Go application.
+
+For instructions on how to deploy this contract to Ethereum from a Go 
+native application read our [Go bindings page](/docs/dapp/native.md).
```
Comment (jmcook1186) [inline docs/_dapp/abigen.md:168]:
```suggestion
native application read our [Go bindings page](/docs/dapp/native.md). Note that the contract bytecode is required in addition to the ABI to deploy the contract.
```

File: docs/_dapp/abigen.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+$ go build ./cmd/abigen
+```
+
+## What is an ABI?
+
+Ethereum smart contracts have a schema that defines its functions and return types 
+in the form of a JSON file. This JSON file is known as an _Application Binary Interface_, 
+or ABI. The ABI acts as a specification for precisely how to encode data sent to a 
+contract and how to decode the data the contract sends back. The ABI is the only 
+essential piece of information required to generate Go bindings. Go developers can then
```
Comment (rjl493456442) [inline docs/_dapp/abigen.md:26]: Actually, binary code generated by compiler is already required in cases that binding with Deployment function is expected.

File: docs/_dapp/abigen.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+contract and how to decode the data the contract sends back. The ABI is the only 
+essential piece of information required to generate Go bindings. Go developers can then 
+use the bindings to interact with the contract from their Go application without having 
+to deal directly with data encoding and decoding. An ABI is generated when a contract 
+is compiled.
+
+### Generating the bindings
+
+To demonstrate the binding generator a contract is required. The contract `Storage.sol` 
+implements two very simple functions: `store` updates a user-defined `uint256` to the
```
Comment (rjl493456442) [inline docs/_dapp/abigen.md:34]: perhaps `user-provided`?

`uint256` is builtin type, feels confused about `user-defined`

**Comment by rjl493456442** [comment]
It's really nice! But I would be appreciate if we can have this information as well

- abigen can also accept the `combined-json` file generated by compiler, with abi and binary code included, by `--combined-json` flag

File: docs/_dapp/abigen.md (hunk 3 above)
Comment (jmcook1186) [reply docs/_dapp/abigen.md:34]:
```suggestion
implements two very simple functions: `store` updates a user-provided `uint256` to the 
```

File: docs/_dapp/abigen.md (hunk 2 above)
Comment (jmcook1186) [reply docs/_dapp/abigen.md:26]:
```suggestion
essential piece of information required to generate Go bindings, unless a deployment function is expected in which case the contrat bytecode is also needed. Go developers can then 
```

File: docs/_dapp/abigen.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+	StorageFilterer   // Log filterer for contract events
+}
+...
+
+```
+
+`Storage.go` contains all the bindings required to interact with 
+`Storage.sol` from a Go application.
+
+For instructions on how to deploy this contract to Ethereum from a Go
```
[the remark this answers is not in the document]
Comment (jmcook1186) [reply docs/_dapp/abigen.md:167]: done!

**Pull request merged**